### PR TITLE
Registration status lookup urls

### PR DESCRIPTION
**Pull request by theJohnnyBrown**
Adds two fields to the voter registration authority to store the URL and shortened URL where voters can look up whether they are registered. There is a [new election authorities spreadsheet with these URLs included](https://docs.google.com/spreadsheets/d/110U-CP48oIL3-3G5U_ZcadAzOd6Pih1KrjP0sVMg9Eo/edit?usp=sharing)

Also adds a small test confirming that the parsed spreadsheet row will validate in the handler.

### CODE REVIEW

File: src/election_authority_works/entities/authority.clj [hunk 1]
```diff
@@ -93,7 +103,10 @@
 
 (defn ocd->tx-map
   [[ocd-division-id description {:keys [voter-registration-methods
+                                        voter-registration-status-lookup
```
Comment (ddellacosta) [inline src/election_authority_works/entities/authority.clj]: Looks like this got removed from the schema?

File: src/election_authority_works/import/core.clj [hunk 2]
```diff
@@ -22,16 +22,21 @@
   [{:keys [primary-voting-method-type reg-by-mail-race-field-req] :as row}]
   (cond-> {}
     (not (str/blank? primary-voting-method-type))
-    (assoc :primary-voting-method-type (edn/read-string primary-voting-method-type))
+    (assoc :primary-voting-method-type
```
Comment (ddellacosta) [inline src/election_authority_works/import/core.clj]: Is the reformatting/adding spaces here, below, and in other namespaces necessary?  I would argue that, in general, outside of conforming to established conventions, we should avoid adding or removing indentation, spaces, and the like when it's not in the code we are working on, as it can turn into a back-and-forth of tweaking stuff to suit our personal preferences (I doubt you were doing that to be clear, I know a lot of the time these just happen when editing code, without thinking), and it makes it harder to figure out what is meaningful in a code review.

File: test/election_authority_works/import/core_test.clj [hunk 3]
```diff
@@ -59,12 +61,6 @@
    :ovr-instructions-es "OVR Instructions - Spanish"
    :ovr-iframe "FALSE"
 
-   ;; Not Applicable/Election Day Registration
```
Comment (ddellacosta) [inline test/election_authority_works/import/core_test.clj]: Why were these removed?

File: test/election_authority_works/import/core_test.clj [hunk 4]
```diff
@@ -180,3 +180,6 @@
               {:keys [ocd-division]} nys-authority
               ocd-division-map (nth ocd-division 2)]
           (is (nil? (get ocd-division-map :voting-methods))))))))
+
+(deftest parsed-data-is-valid
+  (s/validate handlers/create-schema (import/parse-row nys-csv-data-fixture)))
```
Comment (ddellacosta) [inline test/election_authority_works/import/core_test.clj]: Is this really an appropriate test for this ns? (I'm asking very sincerely, I'm really not sure...)

File: src/election_authority_works/entities/authority.clj [hunk 5]
```diff
@@ -80,6 +80,16 @@
     :voter-registration-method/acceptable-form
     :voter-registration-method/instruction]))
 
+(defn maybe-url [url] (when url (java.net.URI. url)))
+(defn voter-registration-status-lookup->tx-map
+  [{:keys [url url-shortened]}]
+  (tx/dissoc-optional-nil-values
+   {:voter-registration-status/url (when url (java.net.URI. url))
```
Comment (ddellacosta) [inline src/election_authority_works/entities/authority.clj]: I think you want to use the `maybe-url` function defined above here, right? Also, I would suggest using `string/blank?` vs. just a `nil` check in that function to confirm we are also disallowing empty strings.

File: test/election_authority_works/import/core_test.clj (hunk 3 above)
Comment (theJohnnyBrown) [reply test/election_authority_works/import/core_test.clj]: The schema specifies that if there is a `:not-applicable` voter registration method, it must be the only one

File: test/election_authority_works/import/core_test.clj (hunk 4 above)
Comment (theJohnnyBrown) [reply test/election_authority_works/import/core_test.clj]: I think so, the thing being tested is `election-authority-works.import.core/parse-row`

File: src/election_authority_works/import/core.clj (hunk 2 above)
Comment (theJohnnyBrown) [reply src/election_authority_works/import/core.clj]: Noted

File: test/election_authority_works/import/core_test.clj (hunk 3 above)
Comment (theJohnnyBrown) [reply test/election_authority_works/import/core_test.clj]: The comment should maybe have stayed

File: src/election_authority_works/entities/authority.clj (hunk 5 above)
Comment (theJohnnyBrown) [reply src/election_authority_works/entities/authority.clj]: Good catch

File: src/election_authority_works/entities/authority.clj (hunk 1 above)
Comment (theJohnnyBrown) [reply src/election_authority_works/entities/authority.clj]: 👍

File: test/election_authority_works/import/core_test.clj (hunk 4 above)
Comment (ddellacosta) [reply test/election_authority_works/import/core_test.clj]: That's reasonable, I guess the thing that is throwing me off is that we are essentially using a completely unrelated ns (I mean, in terms of code-paths-speaking, obviously semantically it's 100% related) to test. On the flip side, it's probably gonna be the most useful test in terms of telling us how correct this all is, practically speaking. ¯\_(ツ)_/¯

File: src/election_authority_works/import/core.clj (hunk 2 above)
Comment (ddellacosta) [reply src/election_authority_works/import/core.clj]: To be clear: I'm trying to open this up for discussion, I don't think we have a meta-convention here...I don't know that everyone is going to agree with my take so feel free to argue back!

File: test/election_authority_works/import/core_test.clj (hunk 3 above)
Comment (ddellacosta) [reply test/election_authority_works/import/core_test.clj]: It may even be handy to add a comment exactly what you just responded to me with--that was helpful.

File: src/election_authority_works/import/core.clj (hunk 2 above)
Comment (ddellacosta) [reply src/election_authority_works/import/core.clj]: (If you even care...feel free to _not_ let me waste your time bikeshedding...)

File: src/election_authority_works/entities/authority.clj (hunk 5 above)
Comment (ddellacosta) [reply src/election_authority_works/entities/authority.clj]: Oh looks like you weren't even using it huh! Removing the whole function is even better...
